receiver/dcgmreceiver/util.go

```diff
@@ -39,44 +48,44 @@ func (m *dcgmMetric) asInt64() int64 {
 	return *(*int64)(unsafe.Pointer(&m.value[0]))
```

Comment:
As the name implies, this is unsafe. Accessing a pointer in this way is undefined behavior in Go.

You want `int64(binary.NativeEndian.Uint64(m.value[:8]))`, or for processing a whole slice at a time, `binary.Read(...)`. Ditto for the three functions above.

Reply:
Found an even more robust solution. There is no more usage of `unsafe` in this receiver. Can't speak for [`go-dcgm`](https://github.com/NVIDIA/go-dcgm/blob/f83cdef499b2cbaba53ab36257e492cfbac0ab60/pkg/dcgm/fields.go#L185-L191), though…